TF_main_loop/validate.py

```diff
@@ -5,26 +5,29 @@
 
 import gflags
 from tqdm import tqdm
+import tensorflow as tf
 
 from main_utils import compute_chunk_size
 
 
 # Print prediction hotmap
-def save_heatmap_fn(x, of, y_pred, labels, save_basedir, subset, f):
+def save_heatmap_fn(x, of, y_soft_pred, labels, nclasses, save_basedir,
```

Comment:
You can probably use `cfg.parameter` instead of passing these new parameters as argument here as well.